omnigibson/systems/macro_particle_system.py

```diff
@@ -1189,6 +1192,12 @@ def get_particles_position_orientation(cls):
             pos, ori = np.array([]).reshape(0, 3), np.array([]).reshape(0, 4)
         return pos, ori
 
+    @classmethod
+    def update(cls):
+        if cls.has_particles and not cls.has_refreshed:
+            cls._refresh_particles_view()
+            cls.has_refreshed = True
+
```

Comment:
This is a bug fix with @ChengshuLi . Basically we need to refresh particles view after a physics step after the first particle has been created, otherwise we will get cannot retrieve rigidview backend error when trying to query transforms of the particles.

Reply:
removed